Firestore/core/src/firebase/firestore/remote/write_stream.cc

```diff
@@ -136,9 +139,17 @@
     // write itself might be causing an error we want to back off from.
     backoff_.Reset();
 
-    callback_->OnWriteStreamMutationResult(
-        serializer_bridge_.ToCommitVersion(response),
-        serializer_bridge_.ToMutationResults(response));
+    auto maybe_version = write_serializer_.ToCommitVersion(*response);
+    if (!maybe_version.ok()) {
```

Comment:
This kind of clutter was exactly what I had hoped to avoid by introducing the `Reader` previously. 

Also, shouldn't serializer operations be `Encode` and `Decode` to clearly indicate which direction the transformation is going?

```c++
Reader reader;
auto version = write_serializer_.DecodeCommitVersion(&reader, *response);
auto results = write_serializer_.DecodeMutationResults(&reader, *response);
if (!reader.ok()) return reader.status();
callback_->OnWriteStreamMutationResult(version, results);
```

Reply:
Parse for bytes to proto seems reasonable.